src/main/java/hudson/plugins/active_directory/ActiveDirectoryDomain.java

```diff
@@ -197,6 +204,13 @@ public TlsConfiguration getTlsConfiguration() {
         return tlsConfiguration;
     }
 
+    private Object readResolve() throws ObjectStreamException {
```

Comment:
Personally I don't like the method to be private. It is executed by reflection, so this means a violation of encapsulation. Also the throws is not needed
```suggestion
    protected Object readResolve() {
```